nodebuilder/blobstream/types.go

```diff
@@ -0,0 +1,186 @@
+package blobstream
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/nmt/namespace"
+	"github.com/tendermint/tendermint/crypto/merkle"
+
+	"github.com/tendermint/tendermint/libs/bytes"
+	"github.com/tendermint/tendermint/types"
+)
+
+// ResultDataCommitment is the API response containing a data
+// commitment, aka data root tuple root.
+type ResultDataCommitment struct {
+	DataCommitment bytes.HexBytes `json:"data_commitment"`
+}
+
+// ResultDataRootInclusionProof is the API response containing the binary merkle
+// inclusion proof of a height to a data commitment.
+type ResultDataRootInclusionProof struct {
+	Proof merkle.Proof `json:"proof"`
+}
+
+// ResultShareProof is the API response that contains a ShareProof.
+// A share proof is a proof of a set of shares to the data root.
+type ResultShareProof struct {
+	ShareProof types.ShareProof `json:"share_proof"`
+}
+
+// ResultCommitmentProof is an API response that contains a CommitmentProof.
+// A commitment proof is a proof of a blob share commitment to the data root.
+type ResultCommitmentProof struct {
+	CommitmentProof CommitmentProof `json:"commitment_proof"`
+}
+
+// CommitmentProof is an inclusion proof of a commitment to the data root.
+// TODO: Ask reviewers if we need protobuf definitions for this
```

Comment:
[note to reviewers]
do we need to create protobuf definitions for these types? 
similarly, do we want json marshellers/unmarshallers?

Reply:
I think protobuf definitions is a good call! Would be useful for consuming and sending to e.g. on-chain lightclient via std cosmos-sdk grpc apis for something like ibc. 

However, it might be good to consider where these protobufs are maintained. Similarly, I see this PRs consumes a number of pkgs from celestia-app. I think solving https://github.com/celestiaorg/go-square/issues/55 would be a step towards further reducing the dependency tree - as I assume ideally, node should not rely on app.

Reply:
makes sense 👍  ideally, we have the types defined in a single place

Reply:
[350d950](https://github.com/celestiaorg/celestia-node/pull/3470/commits/350d9503b3289cd9f939eab2551414686d5e115a)